vector/src/Data/Vector.hs

```diff
@@ -2075,7 +2075,7 @@ unsafeFreeze :: PrimMonad m => MVector (PrimState m) a -> m (Vector a)
 unsafeFreeze = G.unsafeFreeze
 
 -- | /O(1)/ Unsafely convert an immutable vector to a mutable one without
--- copying. The immutable vector may not be used after this operation.
+-- copying. The immutable vector may not be used after this operation if the thawed vector is mutated.
```

Comment:
This double-conditioned sentence is quite entangled to parse. But besides stylistic nitpicking, I think it's valuable to explain "why?", what is the causal relationship here? I suggest something along the following lines:

```
Any mutation of the result (e. g., 'write') corrupts the input vector, breaking the promise of referential transparency. Thus the input must not be accessed after 'unsafeThaw', unless the output is used solely for reading (e. g., by 'copy').
```

Reply:
I don't follow where the double-condition is in my sentence (there's only one `if` in it), but I do like your additional explanation.